packages/web/src/types.ts

```diff
@@ -0,0 +1,190 @@
+export type components = string[];
+
+export type children = any;
+
+export type data = any[];
+
+export type dataFieldArray =
+	| string
+	| string[];
+
+export interface dataNumberBox {
+	label?: string;
+	start: number;
+	end: number;
+}
+
+export type date =
+	| string
+	| string[];
+
+export type dateObject = object;
+
+export interface dateRange {
+	start?:
+	| string
+	| string[];
+	end?:
+	| string
+	| string[];
+}
+
+export type fieldWeights = number[];
+
+export type filterLabel = string;
+
+export type fuzziness =
+	| 0
+	| 1
+	| 2
+	| "AUTO";
+
+export type headers = object;
+
+export type hits = object[];
+
+export type iconPosition =
+	| "left"
+	| "right";
+
+export type labelPosition =
+	| "left"
+	| "right"
+	| "top"
+	| "bottom";
+
+export type options =
+	| object
+	| object[];
+
+export type paginationAt =
+	| "top"
+	| "bottom"
+	| "both";
+
+export interface range {
+	start?: number;
```

Comment:
`start` and `end` are required keys

Reply:
Mirrored from core/types.js
Should I make it required at both places?

Reply:
Yeah